finesse/gui/interferometer_monitor.py

```diff
@@ -0,0 +1,204 @@
+
+    def __init__(self, prop_labels: list[str], prop_units: list[Optional[str]]) -> None:
+        """Creates a sequence of widgets to view properties monitored by the EM27."""
+        super().__init__("EM27 SOH Monitor")
+
+        self._prop_labels = deepcopy(prop_labels)
+        self._prop_units = deepcopy(prop_units)
+
+        self._psf27_temp_box = QLineEdit(self._prop_units[0])
+        self._psf27_temp_box.setReadOnly(True)
+        self._psf27_temp_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._cryo_temp_box = QLineEdit(self._prop_units[1])
+        self._cryo_temp_box.setReadOnly(True)
+        self._cryo_temp_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._bb_hum_box = QLineEdit(self._prop_units[2])
+        self._bb_hum_box.setReadOnly(True)
+        self._bb_hum_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._src_temp_box = QLineEdit(self._prop_units[3])
+        self._src_temp_box.setReadOnly(True)
+        self._src_temp_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._aux_volt_box = QLineEdit(self._prop_units[4])
+        self._aux_volt_box.setReadOnly(True)
+        self._aux_volt_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._aux_current_box = QLineEdit(self._prop_units[5])
+        self._aux_current_box.setReadOnly(True)
+        self._aux_current_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._laser_current_box = QLineEdit(self._prop_units[6])
+        self._laser_current_box.setReadOnly(True)
+        self._laser_current_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._poll_light = LEDIcon.create_poll_icon()
+
+        layout = self._create_controls()
+
+        self._poll_light.setSizePolicy(self._laser_current_box.sizePolicy())
+
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the widgets for the EM27 properties.
+
+        Returns:
+            QGridLayout: The layout with the widgets.
+        """
+        layout = QGridLayout()
+
+        # Add labels for properties to monitor
+        for i, label in enumerate(self._prop_labels):
+            layout.addWidget(QLabel(label), i, 0)
+
+        # Add boxes to show monitored properties' values
+        layout.addWidget(self._psf27_temp_box, 0, 1)
+        layout.addWidget(self._cryo_temp_box, 1, 1)
+        layout.addWidget(self._bb_hum_box, 2, 1)
+        layout.addWidget(self._src_temp_box, 3, 1)
+        layout.addWidget(self._aux_volt_box, 4, 1)
+        layout.addWidget(self._aux_current_box, 5, 1)
+        layout.addWidget(self._laser_current_box, 6, 1)
+        layout.addWidget(self._poll_light, 7, 1)
+
+        return layout
+
+    def set_psf27_temp(self, val: tuple[str, float]) -> None:
+        """Sets the PSF27 temperature text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._psf27_temp_box.setText(f"{val[0]:.6f} {val[1]}")
```

Comment:
Again, these functions are v boilerplate-y, so I'd figure out a way to do all this without repeating yourself (i.e. so you can just update the relevant text field from the name given to you by the API).